pipeline/psth.py

```diff
@@ -0,0 +1,579 @@
+import logging
+import hashlib
+
+from functools import partial
+from inspect import getmembers
+from itertools import repeat
+import numpy as np
+import datajoint as dj
+import scipy.stats as sc_stats
+
+from . import (lab, experiment, ephys)
+[lab, experiment, ephys]  # NOQA
+
+from . import get_schema_name, dict_to_hash
+from .util import _get_units_hemisphere
+
+schema = dj.schema(get_schema_name('psth'))
+log = logging.getLogger(__name__)
+
+
+@schema
+class TrialCondition(dj.Lookup):
+    '''
+    TrialCondition: Manually curated condition queries.
+
+    Used to define sets of trials which can then be keyed on for downstream
+    computations.
+    '''
+
+    definition = """
+    trial_condition_name:       varchar(128)     # user-friendly name of condition
+    ---
+    trial_condition_hash:       varchar(32)     # trial condition hash - hash of func and arg
+    unique index (trial_condition_hash)
+    trial_condition_func:       varchar(36)     # trial retrieval function
+    trial_condition_arg:        longblob        # trial retrieval arguments
+    """
+
+    @property
+    def contents(self):
+        contents_data = [
+            {
```

Comment:
@ksseverson57 , can you see with Wenxi what PSTH she'd want for her task?

Reply:
```suggestion
    def contents(self):
        contents_data = [
            {   'trial_condition_name': 'touch_protraction',
                'trial_condition_func': '_get_trials_exclude_stim',
                'trial_condition_arg': {
                    'task': 'hf pole loc',
                    'task_protocol': 1,
                    'outcome': 'hit',
                    'early_lick': 'no early',
                    'auto_water': 0,
                    'free_water': 0}
            },
            {   'trial_condition_name': 'touch_retraction',
                'trial_condition_func': '_get_trials_exclude_stim',
                'trial_condition_arg': {
                    'task': 'hf pole loc',
                    'task_protocol': 1,
                    'outcome': 'hit',
                    'early_lick': 'no early',
                    'auto_water': 0,
                    'free_water': 0}
            }
```
Need to add similar conditions for other tasks